scipy/optimize/_linprog_util.py

```diff
@@ -840,10 +853,21 @@ def where(A):
     if rr and A_eq.size > 0 and rank < A_eq.shape[0]:
         warn(redundancy_warning, OptimizeWarning, stacklevel=3)
         dim_row_nullspace = A_eq.shape[0]-rank
-        if dim_row_nullspace <= small_nullspace:
-            A_eq, b_eq, status, message = _remove_redundancy(A_eq, b_eq)
-        if dim_row_nullspace > small_nullspace or status == 4:
-            A_eq, b_eq, status, message = _remove_redundancy_dense(A_eq, b_eq)
+        if rr_method is None:
+            if dim_row_nullspace <= small_nullspace:
+                A_eq, b_eq, status, message = _remove_redundancy(A_eq, b_eq)
+            if dim_row_nullspace > small_nullspace or status == 4:
+                A_eq, b_eq, status, message = _remove_redundancy_dense(A_eq, b_eq)
+        else:
+            rr_method = rr_method.lower()
+            if rr_method == "svd":
+                A_eq, b_eq, status, message = _remove_redundancy(A_eq, b_eq)
+            elif rr_method == "pivot":
+                A_eq, b_eq, status, message = _remove_redundancy_dense(A_eq, b_eq)
+            elif rr_method == "id":
+                pass
+            else:  # shouldn't get here; option validity checked above
+                pass
```

Comment:
I don't see how these lines have any impact. Perhaps you missed adding ``_remove_redundancy_id(...)``?

Reply:
Maybe you were only looking at the first commit?